src/stripe-payment.js

```diff
@@ -0,0 +1,220 @@
+          grid-area: submit/stripe/stripe/submit;
+        }
+        stripe-elements {
+          grid-area: stripe;
+        }
+        mwc-button {
+          grid-area: submit;
+        }
+        json-viewer {
+          grid-area: output;
+        }
+
+        stripe-elements:not(:defined),
+        json-viewer:not(:defined),
+        mwc-button:not(:defined) {
+          display: none;
+        }
+      `
+    ];
+  }
+  /**
+   * LitElement / popular convention
+   */
+  static get properties() {
+    return {
+      publishableKey: { type: String, attribute: "publishable-key" },
+      shippingOptions: { type: Array },
+      displayItems: { type: Array },
+      amount: { type: Number },
+      label: { type: String },
+      country: { type: String },
+      currency: { type: String },
+      paymentMethod: { type: Object },
+      error: { type: Object },
+      output: { type: Object },
+      unsupported: { type: Boolean },
+      submitDisabled: { type: Boolean },
+      paymentText: { type: String, attribute: "payment-text" },
+      debug: { type: Boolean }
+    };
+  }
+  /**
+   * HTMLElement
+   */
+  constructor() {
+    super();
+    this.submitDisabled = true;
+    this.debug = false;
+    this.displayItems = [];
+    this.shippingOptions = [];
+    this.amount = 0;
+    this.paymentText = "Submit";
+    this.label = "Purchase";
+    this.country = "US";
+    this.currency = "usd";
+    import("@power-elements/stripe-elements/stripe-payment-request.js");
+  }
+  /**
+   * LitElement render
+   */
+  render() {
+    return html`
+      <stripe-payment-request
+        ?hidden="${this.output || this.unsupported}"
+        publishable-key="${ifDefined(
+          this.publishableKey ? this.publishableKey : undefined
+        )}"
+        .shippingOptions="${this.shippingOptions}"
+        .displayItems="${this.displayItems}"
+        amount="${this.amount}"
+        label="${this.label}"
+        country="${this.country}"
+        currency="${this.currency}"
+        @payment-method="${this.onPaymentMethod}"
+        @error="${this.onError}"
+        @success="${this.onSuccess}"
+        @fail="${this.onFail}"
+        @ready="${this.onReady}"
+        @unsupported="${this.onUnsupported}"
+        generate="source"
+        request-payer-name
+        request-payer-email
+        request-payer-phone
+      >
+        ${this.displayItems.map(
+          item => html`
+            <stripe-display-item
+              data-amount="${item.amount}"
+              data-label="${item.label}"
+            ></stripe-display-item>
+          `
+        )}
+        ${this.shippingOptions.map(
+          ship => html`
+            <stripe-shipping-option
+              data-id="${ship.id}"
+              data-label="${ship.label}"
+              data-detail="${ship.detail}"
+              data-amount="${ship.amount}"
+            ></stripe-shipping-option>
+          `
+        )}
+      </stripe-payment-request>
+      ${this.debug
```

Comment:
I think i'd rather leave this to the consumer, rather than implement `<json-viewer>` internally